drools-parser/src/main/java/org/drools/parser/DRLVisitorImpl.java

```diff
@@ -64,6 +65,8 @@ private void applyChildrenDescrs(PackageDescr packageDescr, List<BaseDescr> desc
                 packageDescr.addGlobal((GlobalDescr) descr);
             } else if (descr instanceof FunctionImportDescr) {
```

Comment:
@tkobayas 
I think it would be nicer to avoid such list of "if/else", and out of curiosity, would it be possible to:
1. extend `BaseDesc`
2. define an abstract method in it that accept as parameter `PackageDescr` (e.g. `executePackageDescr`)
3. any specific implementation (e.g. `GlobalDescr`, `FunctionImportDescr`, ecc) implment it with the snippet that currently is present inside the "if"

The result would be that instead of all those `if descr instance of (....)` there would be just one
`descr.executePackageDescr(packageDescr)`

Is that clear ? Does this make sense ? Would it be possible ?

Reply:
Thank you for the suggestion, @gitgabrio . It makes sense. I filed it as a separate JIRA https://issues.redhat.com/browse/DROOLS-7564 and will work on it (after the drools repo migration will be settled).